comit/src/expiries.rs

```diff
@@ -35,104 +40,140 @@ pub struct Expiries<A, B> {
     alpha_connector: A,
     /// Beta blockchain connector for getting the current time.
     beta_connector: B,
-    /// The alpha ledger relative expiry time.
-    alpha_expiry: Duration,
-    /// The beta ledger relative expiry time.
-    beta_expiry: Duration,
+    /// The alpha ledger expiry offset.
+    alpha_offset: AlphaOffset,
+    /// The beta ledger expiry offset.
+    beta_offset: BetaOffset,
+}
+
+/// Calculate a pair of useful expiries, 'useful' means a swap can be
+/// successfully completed with these expiries.
+/// Multiply the resulting offsets by `scale_factor` percentage to give the
```

Comment:
The logic to create a set of expiries and the logic that checks if they are useful uses the same durations so if we create the expiries and calculate the absolute timestamps from 'now' they immediately go stale (assuming no action is taken). This can be resolved in two ways (at least), one is by adding some time window at the front, we already explored this and it has its own problems, the other I came up with was adding a scaling factor. So now we have both ways available, I'm hoping that with usage we will see if either of these is more correct than the other, perhaps we don't need either, perhaps we need something else. Without actually using the module its hard to tell.